Src/Newtonsoft.Json/Converters/BsonObjectIdConverter.cs

```diff
@@ -33,26 +33,24 @@ namespace Newtonsoft.Json.Converters
     /// <summary>
     /// Converts a <see cref="BsonObjectId"/> to and from JSON and BSON.
     /// </summary>
-    public class BsonObjectIdConverter : JsonConverter
+    public class BsonObjectIdConverter : JsonConverter<BsonObjectId>
```

Comment:
According to https://stackoverflow.com/questions/1456785/a-definitive-guide-to-api-breaking-changes-in-net#1522718 changing the base class won't be a breaking change besides through reflection.